hydrosdk/modelversion.py

```diff
@@ -335,86 +317,87 @@ class ModelStatus(Enum):
     Failed = "Failed"
 
 
-class Model(Metricable):
+class ModelVersion(Metricable):
     """
     Model (A model is a machine learning model or a processing function that consumes provided inputs
     and produces predictions or transformations
     https://hydrosphere.io/serving-docs/latest/overview/concepts.html#models)
+    ModelVersion represents one of the Model's versions.
     """
     BASE_URL = "/api/v2/model"
 
     @staticmethod
-    def find(cluster, name, version):
+    def find_by_name_id(cluster: Cluster, name: str, id_) -> 'ModelVersion':
         """
         Finds a model on server by name and model version
 
         :param cluster: active cluster
         :param name: model name
-        :param version: model version
+        :param id_: model version
         :raises Exception: if server returned not 200
-        :return: Model obj
+        :return: ModelVersion obj
         """
-        resp = cluster.request("GET", Model.BASE_URL + "/version/{}/{}".format(name, version))
+        resp = cluster.request("GET", ModelVersion.BASE_URL + "/version/{}/{}".format(name, id_))
 
         if resp.ok:
             model_json = resp.json()
-            return parse_model_from_json_dict(cluster, model_json)
+            return ModelVersion.from_json(cluster=cluster, model_version=model_json)
 
         else:
             raise Exception(
-                f"Failed to find Model for name={name}, version={version} . {resp.status_code} {resp.text}")
+                f"Failed to find Model for name={name}, version={id_} . {resp.status_code} {resp.text}")
 
     @staticmethod
-    def find_by_id(cluster, model_id):
+    def find_by_id(cluster: Cluster, id_) -> 'ModelVersion':
         """
-        Finds a model on server by id
+        Finds a modelversion on server by id
 
         :param cluster: active cluster
-        :param model_id: model id
+        :param id_: model version id
         :raises Exception: if server returned not 200
-        :return: Model obj
+        :return: ModelVersion obj
         """
-        resp = cluster.request("GET", Model.BASE_URL + "/version")
+        resp = cluster.request("GET", ModelVersion.BASE_URL + "/version")
 
         if resp.ok:
             for model_json in resp.json():
-                if model_json['id'] == model_id:
-                    return parse_model_from_json_dict(cluster, model_json)
+                if model_json['id'] == id_:
+                    return ModelVersion.from_json(cluster=cluster, model_version=model_json)
 
         raise Exception(
```

Comment:
A nice opportunity to specialize modelversion-related exceptions, or to reimplement the method to return None.